cnf-certification-test/platform/nodetainted/nodetainted.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package nodetainted
+
+import (
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/ocpclient"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf/identifier"
+	utils "github.com/test-network-function/cnf-certification-test/pkg/tnf/utils"
+)
+
+// NodeTainted holds information about tainted nodes.
+type NodeTainted struct {
+	result  int
+	timeout time.Duration
+	// args       []string
+	OCPClient  ocpclient.OcpClient
+	OCPContext ocpclient.Context
+}
+
+// NewNodeTainted creates a new NodeTainted tnf.Test.
+func NewNodeTaintedTester(timeout time.Duration, client ocpclient.OcpClient, ctx ocpclient.Context) *NodeTainted {
+	return &NodeTainted{
+		timeout:    timeout,
+		result:     tnf.ERROR,
+		OCPClient:  client,
+		OCPContext: ctx,
+	}
+}
+
+// GetIdentifier returns the tnf.Test specific identifier.
+func (nt *NodeTainted) GetIdentifier() identifier.Identifier {
+	return identifier.NodeTaintedIdentifier
+}
+
+// Timeout returns the timeout in seconds for the test.
+func (nt *NodeTainted) Timeout() time.Duration {
+	return nt.timeout
+}
+
+// Result returns the test result.
+func (nt *NodeTainted) Result() int {
+	return nt.result
+}
+
+func (nt *NodeTainted) runCommand(cmd string) string {
```

Comment:
in here, runCommand should not be the one deciding if there's a success. it should just run command and return the output. This is now behaving like executing a test